cronos-play/getting-started_unreal/quick-start/dynamic-contract.md

```diff
@@ -0,0 +1,132 @@
+# Dynamic Contract 
+
+with Dynamic Contract, you can call any smartcontract. you need abi json file for the contract.  
+- create contract beforehand, prepare contract address 
+- prepare abi json file for the contract to access
+
+## DynamicContractActor apis
+### property variables
+- defiWallet : connect to DefiWalletCoreActor for signing
+### constructor
+- NewEthContract : for querying contract
+- NewSigningEthContract : for modifying contract
+### functions
+- EncodeDynamicContract : for encoding
+- CallDynamicContract : for calling
+- SendDynamicContract : for signing and sending
+
+
+## Call
+this is for non modifying contract call
+
+<figure><img src="../../../.gitbook/assets/cronos-gamefi-blueprint-dynamic-contract-call.png" alt=""><figcaption></figcaption></figure>
+
+
+
+## Send
+this is for modifying contract call. it needs `DefiWalletCoreActor` to sign the transaction.
+
+<figure><img src="../../../.gitbook/assets/cronos-gamefi-blueprint-dynamic-contract-send.png" alt=""><figcaption></figcaption></figure>
+
+
+
+## Encode
+this is for wallet-connect. it generates data which can be included in the transaction. and the transaction can be signed with wallet-connect
+<figure><img src="../../../.gitbook/assets/cronos-gamefi-blueprint-dynamic-contract-encode.png" alt=""><figcaption></figcaption></figure>
+
+
+## For function arguments encoding
+### Sample for safeMint
+ - solidity function signature:
+ ```
+ safeMint(address,string)
+ ```
+- function name: 
+```
+safeMint
+```
+- function arguments: 
+```
+[{"Address":{"data":"0x00"}},{"Str":{"data":"my"}}]
```

Comment:
is this placeholder data?

Reply:
not placeholder.
this is for json encoding for function argument.

Reply:
it's given as json, and restored as solidity arguments

Reply:
Ok